gnovm/pkg/gnolang/preprocess.go

```diff
@@ -3073,6 +3073,237 @@ func findUndefined(store Store, last BlockNode, x Expr) (un Name) {
 	return findUndefined2(store, last, x, nil)
 }
 
+// finds the next undefined identifier and returns it if it is global
+func findUndefined2SkipLocals(store Store, last BlockNode, x Expr, t Type) Name {
+	name := findUndefined2(store, last, x, t)
+
+	existsLocal := func(name Name, bn BlockNode) bool {
```

Comment:
`GetPathForName` is behind an interface and it would cause a lot more code changes to modify that. I think, having  a local function encapsulates this custom logic and other parts of the codebase don't need to know about it.

Reply:
ok to me